cmd/common/image/build.go

```diff
@@ -320,6 +374,21 @@ func buildInContainer(opts *cmdOpts, conf *buildFile) error {
 		return err
 	}
 
+	imgOpts := &oci.ImageOptions{
+		VerifyOptions: oci.VerifyOptions{
+			VerifyPublicKey: opts.verifyImage,
+			PublicKey:       opts.publicKey,
+		},
+		AuthOptions: oci.AuthOptions{
+			AuthFile: oci.DefaultAuthFile,
+		},
```

Comment:
Is is this part needed?

Reply:
I don't think `oci.DefaultAuthFile` is right, these are the credentials to pull the builder image, and `oci.DefaultAuthFile` should store credentials to pull gadgets, which could be totally different. What happens if it's left empty to fallback to the docker credentials?

https://github.com/inspektor-gadget/inspektor-gadget/blob/8f7cb770488817dc5e51d56720761f30e95ea5a4/pkg/oci/oci.go#L523-L534

Reply:
Alright, the logic there checks if AuthFile is DefaultAuthFile:

```golang
		// If the AuthFile was not set explicitly, we allow to fall back to the docker auth,
		// otherwise we fail to avoid masking an error from the user
		if !errors.Is(err, os.ErrNotExist) || authOptions.AuthFile != DefaultAuthFile {
			return nil, fmt.Errorf("opening auth file %q: %w", authOptions.AuthFile, err)
		}
```

IMO there should also be some logic to fallback to the default auth file if it's not set, but I don't want to deviate this PR, so using `AuthFile: oci.DefaultAuthFile,` is fine for me. Perhaps just add a comment saying it's needed by the library in order to work or something like that.